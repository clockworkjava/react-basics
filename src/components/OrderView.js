import React from 'react';

export default class OrderView extends React.Component {

 

    render() {
        return (
            <div className="orderView">
                <span>{this.props.book.name}</span>
                <button className="btn btn-danger" onClick={ (event) => this.props.removeFromOrder(this.props.book.name)}>Remove</button>
            </div>
                
        );
    }

}