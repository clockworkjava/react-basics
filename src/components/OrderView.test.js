import React from 'react';
import ReactDOM from 'react-dom';
import OrderView from './OrderView';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Order view tests', () => {

    it('Snapshot matches', () => {
        const book = {
            name: "Book title"
        }
        const wrapper = shallow(<OrderView book={book}/>);
        expect(wrapper).toMatchSnapshot();
    })

    it('Title of the book is displayed as in props', () => {
        const book = {
            name: "Book title"
        }
        const wrapper = shallow(<OrderView book={book}/>);
        expect(wrapper.find('.orderView').find('span').text()).toBe(book.name);
    })

})

