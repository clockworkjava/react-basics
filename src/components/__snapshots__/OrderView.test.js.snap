// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order view tests Snapshot matches 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderView
    book={
        Object {
            "name": "Book title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          Book title
</span>,
        <button
          className="btn btn-danger"
          onClick={[Function]}
>
          Remove
</button>,
      ],
      "className": "orderView",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Book title",
        },
        "ref": null,
        "rendered": "Book title",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove",
          "className": "btn btn-danger",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Book title
</span>,
          <button
            className="btn btn-danger"
            onClick={[Function]}
>
            Remove
</button>,
        ],
        "className": "orderView",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Book title",
          },
          "ref": null,
          "rendered": "Book title",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove",
            "className": "btn btn-danger",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
